test/unit/math/rev/arr/functor/coupled_ode_system_test.cpp

```diff
@@ -39,15 +41,21 @@ TEST_F(StanAgradRevOde, coupled_ode_system_dv) {
   z0.push_back(1.0);
   z0.push_back(2.0);
 
+  std::size_t stack_size = stan::math::nested_size();
+
   coupled_ode_system<harm_osc_ode_fun, double, stan::math::var> system(
       harm_osc, y0, theta, x, x_int, &msgs);
 
+  EXPECT_EQ(stan::math::nested_size(), stack_size);
```

Comment:
What exactly is this testing? It looks like you're checking to see that the constructor doesn't add anything to the autodiff stack. If that's it, for clarity, could you add a note to the test? Something like:

```
EXPECT_EQ(stack_size, stan::math::nested_size()) << "expecting no new things on the stack";
```

(And switch the order of the arguments. In Google Test, the first is what you expect to be true, the second is what is being tested.)

Reply:
Additional question: is there a reason you didn't do this with just the stack size instead? Why use the nested autodiff?

Reply:
Yes, I am testing for the stack size not increasing by the constructor. I think you asked for this a while ago.

I am happy to add that cout message and swap order (I did not know about order here).

I used nested AD, because there is no way to get the stack size of the AD tape which is not nested; at least the last time I checked.

Reply:
Thanks for the answers! That makes sense. And it does make sense to check for this... we don't want a future implementation to just stomp on this behavior accidentally.